src/main/java/com/amplitude/Amplitude.java

```diff
@@ -135,16 +138,27 @@ public synchronized void flushEvents() {
     if (eventsToSend.size() > 0) {
       List<Event> eventsInTransit = new ArrayList<>(eventsToSend);
       eventsToSend.clear();
-      CompletableFuture.supplyAsync(
-          () -> {
-            Response response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
-            Status status = response.status;
-
-            if (Retry.shouldRetryForStatus(status)) {
-              Retry.sendEventsWithRetry(eventsInTransit, response, httpCall);
-            }
-            return null;
-          });
+      CompletableFuture<Response> asyncHandler =
+          CompletableFuture.supplyAsync(
+              () -> {
+                Response response = null;
+                try {
+                  response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
+                } catch (AmplitudeInvalidAPIKeyException e) {
+                  e.printStackTrace();
+                }
+                return response;
+              });
+
+      try {
+        Response response = asyncHandler.join();
+        Status status = response.status;
+        if (Retry.shouldRetryForStatus(status)) {
+          Retry.sendEventsWithRetry(eventsInTransit, response, httpCall);
+        }
+      } catch (CompletionException e) {
+        e.printStackTrace();
```

Comment:
I mean, ideally we may want this method `flushEvents` just throw the Invalid api key exception to let user aware. And also, we don't really want to print stack trace in our code.